### PR TITLE
feat: add Nightwatch for end-to-end testing

### PR DESCRIPTION
**Pull request by vaibhavsingh97**
Hello 👋 ,

This PR integrates [`nightwatch`](https://nightwatchjs.org/) as an end-to-end testing solution for the create-vue CLI. With this integration, developers can quickly write and run automated tests for their Vue applications using the popular nightwatch testing framework.

Thank you

cc @sodatea

### CODE REVIEW

**Comment by vaibhavsingh97** [comment]
I don't know why the tests are failing in CI, locally the tests are passing.

File: template/tsconfig/nightwatch/nightwatch/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  "compilerOptions": {
+    "target": "ESNext",
+    "module": "commonjs",
+    "composite": true,
+    "isolatedModules": false,
+    "preserveValueImports": false,
+    "lib": ["ESNext", "dom"],
+    "types": ["nightwatch"]
+  },
+  "include": ["../node_modules/@nightwatch/**/*", "../src/components/**/*", "../tests/e2e/**/*"],
```
[the remark this answers is not in the document]
Comment (vaibhavsingh97) [reply template/tsconfig/nightwatch/nightwatch/tsconfig.json:12]: This configuration covers both end-to-end and component testing.

File: template/tsconfig/nightwatch/nightwatch/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "extends": "@tsconfig/node18/tsconfig.json",
+  "compilerOptions": {
+    "target": "ESNext",
+    "module": "commonjs",
+    "composite": true,
+    "isolatedModules": false,
+    "preserveValueImports": false,
```
[the remark this answers is not in the document]
Comment (vaibhavsingh97) [reply template/tsconfig/nightwatch/nightwatch/tsconfig.json:8]: I had encountered errors previously with TypeScript v5.0, so I disabled it. I learned that vs code wasn't picking up the workspace typescript version, so I don't need it now. Removing it.

**Comment by sodatea** [comment]
Thanks! The templates look pretty good to me now.
However, I still need some more time to play with it a little bit more because of the complexity.

So I'd like to cut one more release of `create-vue` before merging this one.
I plan to ship this feature with the coming update to Vue 3.3, or soon after if the testing takes longer.

**Comment by vaibhavsingh97** [comment]
Hi @sodatea, I was wondering if there are any updates I need to make to this PR for it to be merged.

**Comment by sodatea** [comment]
As far as I've tested, the changes are good.

The only concern is the implicit hard dependency on the `jsx` option, because the jsx plugin is used in the new `vite.config.js`. We want JSX support to be optional.

To fix that issue, we need some kind of logical composition for `vite.config.js` templates, so that the templates can handle more complex scenarios.
But I think that deserves another PR, and I'm merging this one now.

**Review by sodatea — APPROVED** [review]

**Pull request merged**